src/Datadog.Trace.ClrProfiler.Native/util.cpp

```diff
@@ -67,10 +67,17 @@ std::vector<std::wstring> GetEnvironmentValues(const std::wstring &name) {
 
 std::wstring GetCurrentProcessName() {
   std::wstring current_process_path(260, 0);
-  const DWORD len = GetModuleFileName(nullptr, current_process_path.data(),
+  const DWORD len = GetModuleFileName(nullptr, StringData(current_process_path),
                                       (DWORD)(current_process_path.size()));
   current_process_path = current_process_path.substr(0, len);
-  return std::filesystem::path(current_process_path).filename();
+  return std::experimental::filesystem::path(current_process_path).filename();
+}
+
+wchar_t *StringData(const std::wstring &str) {
```

Comment:
Why did we stop using `wstring::data()`? Btw, there's also a `wstring::c_str()` method. Since C++11 they are both guaranteed to be synonyms and to have a null-terminating character, so even for empty strings `&str[0]` is probably valid.